core/src/layers/simulation/real.rs

```diff
@@ -0,0 +1,99 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use crate::ops::{OpList, OpRead, OpScan, OpWrite};
+use crate::raw::{Accessor, Layer, LayeredAccessor, RpList, RpRead, RpScan, RpWrite};
+use async_trait::async_trait;
+
+#[derive(Debug, Copy, Clone, Default)]
+pub struct MadsimLayer;
+
+impl<A: Accessor> Layer<A> for MadsimLayer {
```

Comment:
I feel like we don't need to provide a `real` mod here. It's easier for users to add a layer or not. For example:

```rust
#[cfg(feature = "layers-chaos")]
let op = {
    use opendal::layers::ChaosLayer;
    op.layer(ChaosLayer::new(0.1))
};
```

In `MadsimLayer`, we can focus on madsim itself's logic.